cmd/mtaBuild.go

```diff
@@ -0,0 +1,411 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+)
+
+const templateMtaYml = `_schema-version: "2.0.0"
+ID: "{{.ID}}"
+version: {{.Version}}
+
+parameters:
+  hcp-deployer-version: "1.0.0"
+
+modules:
+  - name: {{.Name}}
+    type: html5
+    path: .
+    parameters:
+       version: {{.Version}}-${timestamp}
+    build-parameters:
+      builder: grunt
+      build-result: dist`
+
+// MTABuildTarget ...
+type MTABuildTarget int
+
+const (
+	// NEO ...
+	NEO MTABuildTarget = iota
+	// CF ...
+	CF MTABuildTarget = iota
+	//XSA ...
+	XSA MTABuildTarget = iota
+)
+
+// ValueOfBuildTarget ...
+func ValueOfBuildTarget(str string) (MTABuildTarget, error) {
+	switch str {
+	case "NEO":
+		return NEO, nil
+	case "CF":
+		return CF, nil
+	case "XSA":
+		return XSA, nil
+	default:
+		return -1, fmt.Errorf("Unknown BuildTarget: '%s'", str)
+	}
+}
+
+// String ...
+func (m MTABuildTarget) String() string {
+	return [...]string{
+		"NEO",
+		"CF",
+		"XSA",
+	}[m]
+}
+
+func mtaBuild(config mtaBuildOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment) {
+	log.Entry().Info("Launching mta build")
+	err := runMtaBuild(config, commonPipelineEnvironment, &command.Command{})
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("failed to execute mta build")
+	}
+
+}
+
+func runMtaBuild(config mtaBuildOptions, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment,
+	e envExecRunner) error {
+
+	e.Stdout(os.Stderr) // keep stdout clear.
+	e.Stderr(os.Stderr)
```

Comment:
please direct logging to logrus, see for example here: https://github.com/SAP/jenkins-library/blob/master/cmd/karmaExecuteTests.go#L16

Reply:
See above. I have still concerns with to log messages like `[INFO][ERROR] build failed`.